node/cmd/guardiand/node.go

```diff
@@ -914,31 +914,23 @@ func runNode(cmd *cobra.Command, args []string) {
 	for _, chainId := range vaa.GetAllNetworkIDs() {
 		chainMsgC[chainId] = make(chan *common.MessagePublication)
 		go func(c <-chan *common.MessagePublication, chainId vaa.ChainID) {
-			zeroAddress := vaa.Address{}
 			for {
 				select {
 				case <-rootCtx.Done():
 					return
 				case msg := <-c:
-					if msg.EmitterChain != chainId {
-						// SECURITY: This should never happen. If it does, a watcher has been compromised.
-						logger.Fatal("SECURITY CRITICAL: Received observation from a chain that was not marked as originating from that chain",
+					shouldPublish, err := validateMessage(logger, msg, chainId)
```

Comment:
I think the code becomes less readable this way, because now there are three different paths in which errors are handled 
1. by setting shouldPublish to false
2. by returning an error
3. by logging an error within validateMessage itself